### ISSUE TITLE
Wyze Cam Pan - ERR: no one is publishing to path 'XXXXX'

### ISSUE THREAD

**Issue by jeffreylin** [mrlt8/docker-wyze-bridge#110]
Getting `[RTMP] ERR: no one is publishing to path 'XXXXX'`

---

`docker-wyze-bridge` seems to connect to the Wyze servers OK and auth properly, but it seems like it's having trouble getting the video feed - you can see the RTMP connection as the last thing in the log, but it doesn't find the video feed despite finding the camera earlier.

Let me know if there's any more info I can provide or things I can try to debug! I tried downgrading to `docker-wyze-bridge:0.5.18` to see if the latest refactor caused some issues, but still got the same results. None of the other video protocols work either.

---

Setup is:
 - Wyze Cam Pan (Firmware 4.10.0.222)
 - Running on RasPi 3B
 - Running `mrlt8/wyze-bridge:latest` which is `v0.6.2` as of now.

**docker-compose.yml**
```
pi@raspberrypi:~/workspace/docker-wyze-bridge $ cat docker-compose.yml
version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: always
        image: mrlt8/wyze-bridge:latest # Use a prebuilt image
        # build:  # Uncomment to build from source
        #     context: ./app # Uncomment to build from source
        #     # dockerfile: Dockerfile.arm # Uncomment to build for arm
        ports:
            - 1935:1935
            - 8554:8554
            - 8888:8888
        environment:
            - WYZE_EMAIL=XXXXX #${WYZE_EMAIL} # Replace with wyze email
            - WYZE_PASSWORD=YYYYYY #${WYZE_PASSWORD} # Replace with wyze password
            - IGNORE_OFFLINE=true
            - DEBUG_LEVEL=debug
            - RTSP_LOGLEVEL=debug
```

**Log**
```
pi@raspberrypi:~/workspace/docker-wyze-bridge $ docker-compose up --build
Starting wyze-bridge ... done
Attaching to wyze-bridge
wyze-bridge    | 2021/09/08 03:26:16 I [0/0] rtsp-simple-server v0.17.3
wyze-bridge    | 2021/09/08 03:26:16 I [0/0] [RTSP] TCP listener opened on :8554
wyze-bridge    | 2021/09/08 03:26:16 I [0/0] [RTMP] listener opened on :1935
wyze-bridge    | 2021/09/08 03:26:16 I [0/0] [HLS] listener opened on :8888
wyze-bridge    | 2021/09/08 03:26:16 I [0/0] [API] listener opened on 127.0.0.1:9997
wyze-bridge    |
wyze-bridge    | 🚀 STARTING DOCKER-WYZE-BRIDGE v0.6.2
wyze-bridge    | 2021/09/08 03:26:17 [wyze_bridge][INFO][MainThread] 📚 Using 'user' from local cache...
wyze-bridge    | 2021/09/08 03:26:17 [wyze_bridge][INFO][MainThread] 📚 Using 'cameras' from local cache...
wyze-bridge    |
wyze-bridge    | 🎬 STARTING ALL 2 CAMERAS
wyze-bridge    | 2021/09/08 03:26:17 [wyze_bridge][DEBUG][Garage Cam] ⌛️ Connecting to cam..
wyze-bridge    | 2021/09/08 03:26:17 [wyze_bridge][DEBUG][Backyard Cam] ⌛️ Connecting to cam..
wyze-bridge    | 2021/09/08 03:26:18 [wyze_bridge][INFO][Backyard Cam] IOTC_ER_DEVICE_OFFLINE
wyze-bridge    | 2021/09/08 03:26:18 [wyze_bridge][INFO][Backyard Cam] 🪦 Camera is offline. Will NOT try again.
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.iotc][INFO][Garage Cam] AV Client Start: chan_id=0 expected_chan=0
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][INFO][Thread-1] Now listening on channel id 0
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10000ConnectRequest code=10000 resp_code=10001> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10000 txt_len=0> b''
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=10001 txt_len=17>: b'\x03\xa0\x83A9\x96M\xa1\xc6\xd5|\xa8\x94\xc8\x9b\x8ay'
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_protocol][DEBUG][Garage Cam] Sending response: <K10002ConnectAuth code=10002 resp_code=10003>
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10002ConnectAuth code=10002 resp_code=10003> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10002 txt_len=22> b'\tua-\xfa\x1c\xce5\xf7\xd3\xb9\x07\x83\xd9vq2CAA\x01\x01'
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=10003 txt_len=642>: b'{"connectionRes":"1","cameraInfo":{"videoParm": {"bitRate":"120","resolution":"1","fps":"15","horizontalFlip":"1","verticalFlip":"1"},"settingParm": {"stateVision":"1", "nightVision":"3", "osd":"1", "logSd":"1", "logUdisk":"1", "telnet":"2","tz":"-7"},"basicInfo": {"firmware":"4.10.0.222","type":"Camera","hardware":"0.0.0.0","model":"WYZECP1_JEF","mac":"MACMACMAC"},"channelResquestResult": {"video":"1","audio":"1"},"recordType": {"type":"1"},"sdParm": {"status":"2","capacity":"0","free":"10"},"uDiskParm": {"status":"2","capacity":"0","free":"0"},"apartalarmParm":{"type":"0","startX":"25","longX":"50","startY":"25","heightY":"50"}}}'
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10056SetResolvingBit code=10056 resp_code=10057> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10056 txt_len=3> b'\x01x\x00'
wyze-bridge    | 2021/09/08 03:26:19 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=1 txt_len=2>: b"H'"
wyze-bridge    | 2021/09/08 03:26:33 I [0/0] [RTMP] [conn 10.10.20.10:58264] opened
wyze-bridge    | 2021/09/08 03:26:33 I [0/0] [path garage-cam] created
wyze-bridge    | 2021/09/08 03:26:33 I [0/0] [RTMP] [conn 10.10.20.10:58264] ERR: no one is publishing to path 'garage-cam'
wyze-bridge    | 2021/09/08 03:26:33 I [0/0] [RTMP] [conn 10.10.20.10:58264] closed
```

**Comment by mrlt8** [comment]
Thanks for the detailed info! I think your firmware was missing `wifidb` which caused the stream to get stuck.

Could you see if v0.6.3 fixes the issue?

**Comment by jeffreylin** [comment]
Wow, that was quick!

Don't see a stream still - here's the new output:

**New Log**
```pi@raspberrypi:~/workspace/docker-wyze-bridge $ docker-compose up --build
Starting wyze-bridge ... done
Attaching to wyze-bridge
wyze-bridge    | 2021/09/08 04:56:03 I [0/0] rtsp-simple-server v0.17.3
wyze-bridge    | 2021/09/08 04:56:03 I [0/0] [RTSP] TCP listener opened on :8554
wyze-bridge    | 2021/09/08 04:56:03 I [0/0] [RTMP] listener opened on :1935
wyze-bridge    | 2021/09/08 04:56:03 I [0/0] [HLS] listener opened on :8888
wyze-bridge    | 2021/09/08 04:56:03 I [0/0] [API] listener opened on 127.0.0.1:9997
wyze-bridge    |
wyze-bridge    | 🚀 STARTING DOCKER-WYZE-BRIDGE v0.6.3
wyze-bridge    | 2021/09/08 04:56:04 [wyze_bridge][INFO][MainThread] 📚 Using 'user' from local cache...
wyze-bridge    | 2021/09/08 04:56:04 [wyze_bridge][INFO][MainThread] 📚 Using 'cameras' from local cache...
wyze-bridge    |
wyze-bridge    | 🎬 STARTING ALL 2 CAMERAS
wyze-bridge    | 2021/09/08 04:56:04 [wyze_bridge][DEBUG][Garage Cam] ⌛️ Connecting to cam..
wyze-bridge    | 2021/09/08 04:56:04 [wyze_bridge][DEBUG][Backyard Cam] ⌛️ Connecting to cam..
wyze-bridge    | 2021/09/08 04:56:05 [wyze_bridge][INFO][Backyard Cam] IOTC_ER_DEVICE_OFFLINE
wyze-bridge    | 2021/09/08 04:56:05 [wyze_bridge][INFO][Backyard Cam] 🪦 Camera is offline. Will NOT try again.
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.iotc][INFO][Garage Cam] AV Client Start: chan_id=0 expected_chan=0
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][INFO][Thread-1] Now listening on channel id 0
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10000ConnectRequest code=10000 resp_code=10001> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10000 txt_len=0> b''
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=10001 txt_len=17>: b'\x03:MM80+S\x8d% \xe0\x8f\xe6\x9c\xe6\xbd'
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_protocol][DEBUG][Garage Cam] Sending response: <K10002ConnectAuth code=10002 resp_code=10003>
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10002ConnectAuth code=10002 resp_code=10003> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10002 txt_len=22> b':\x84,\x15\x0e\x8f\xa6>\x11ufD\x99\x07\xe792CAA\x01\x01'
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=10003 txt_len=642>: b'{"connectionRes":"1","cameraInfo":{"videoParm": {"bitRate":"120","resolution":"1","fps":"15","horizontalFlip":"1","verticalFlip":"1"},"settingParm": {"stateVision":"1", "nightVision":"3", "osd":"1", "logSd":"1", "logUdisk":"1", "telnet":"2","tz":"-7"},"basicInfo": {"firmware":"4.10.0.222","type":"Camera","hardware":"0.0.0.0","model":"WYZECP1_JEF","mac":"MACMACMAC"},"channelResquestResult": {"video":"1","audio":"1"},"recordType": {"type":"1"},"sdParm": {"status":"2","capacity":"0","free":"10"},"uDiskParm": {"status":"2","capacity":"0","free":"0"},"apartalarmParm":{"type":"0","startX":"25","longX":"50","startY":"25","heightY":"50"}}}'
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Garage Cam] SEND <K10056SetResolvingBit code=10056 resp_code=10057> <TutkWyzeProtocolHeader prefix=b'HL' protocol=1 code=10056 txt_len=3> b'\x01x\x00'
wyze-bridge    | 2021/09/08 04:56:07 [wyzecam.tutk.tutk_ioctl_mux][DEBUG][Thread-1] RECV <TutkWyzeProtocolHeader prefix=b'HL' protocol=10 code=1 txt_len=2>: b"H'"
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60782] opened
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60782] [c->s] OPTIONS rtsp://rpi:8554/garage-cam RTSP/1.0
wyze-bridge    | CSeq: 2
wyze-bridge    | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60782] [s->c] RTSP/1.0 200 OK
wyze-bridge    | CSeq: 2
wyze-bridge    | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60782] [c->s] DESCRIBE rtsp://rpi:8554/garage-cam RTSP/1.0
wyze-bridge    | Accept: application/sdp
wyze-bridge    | CSeq: 3
wyze-bridge    | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [path garage-cam] created
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60782] [s->c] RTSP/1.0 404 Not Found
wyze-bridge    | CSeq: 3
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60782] ERR: no one is publishing to path 'garage-cam'
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60782] closed
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60783] opened
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60783] [c->s] SETUP rtsp://rpi:8554/garage-cam RTSP/1.0
wyze-bridge    | CSeq: 0
wyze-bridge    | Transport: RTP/AVP;unicast;client_port=9296-9297
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [session 170066698] opened by 10.10.20.10:60783
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60783] [s->c] RTSP/1.0 400 Bad Request
wyze-bridge    | CSeq: 0
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60783] ERR: path must end with a slash (garage-cam)
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60783] closed
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [session 170066698] closed
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60784] opened
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60784] [c->s] OPTIONS rtsp://rpi:8554 RTSP/1.0
wyze-bridge    | CSeq: 1
wyze-bridge    | Clientchallenge: 9e26d33f2984236010ef6253fb1887f7
wyze-bridge    | Clientid: Linux_2.4_6.0.9.1235_play32_RN01_EN_586
wyze-bridge    | Companyid: KnKV4M4I/B2FjJ1TToLycw==
wyze-bridge    | Guid: 00000000-0000-0000-0000-000000000000
wyze-bridge    | Playerstarttime: [28/03/2003:22:50:23 00:00]
wyze-bridge    | Regiondata: 0
wyze-bridge    | User-Agent: RealMedia Player Version 6.0.9.1235 (linux-2.0-libc6-i386-gcc2.95)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 D [0/0] [RTSP] [conn 10.10.20.10:60784] [s->c] RTSP/1.0 200 OK
wyze-bridge    | CSeq: 1
wyze-bridge    | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:56:12 I [0/0] [RTSP] [conn 10.10.20.10:60784] closed
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60933] opened
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60933] [c->s] OPTIONS rtsp://10.10.20.134:8554/garage-cam RTSP/1.0
wyze-bridge    | CSeq: 2
wyze-bridge    | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60933] [s->c] RTSP/1.0 200 OK
wyze-bridge    | CSeq: 2
wyze-bridge    | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60933] [c->s] DESCRIBE rtsp://10.10.20.134:8554/garage-cam RTSP/1.0
wyze-bridge    | Accept: application/sdp
wyze-bridge    | CSeq: 3
wyze-bridge    | User-Agent: LibVLC/3.0.16 (LIVE555 Streaming Media v2016.11.28)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60933] [s->c] RTSP/1.0 404 Not Found
wyze-bridge    | CSeq: 3
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60933] ERR: no one is publishing to path 'garage-cam'
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60933] closed
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60934] opened
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60934] [c->s] SETUP rtsp://10.10.20.134:8554/garage-cam RTSP/1.0
wyze-bridge    | CSeq: 0
wyze-bridge    | Transport: RTP/AVP;unicast;client_port=9002-9003
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [session 348164318] opened by 10.10.20.10:60934
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60934] [s->c] RTSP/1.0 400 Bad Request
wyze-bridge    | CSeq: 0
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60934] ERR: path must end with a slash (garage-cam)
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60935] opened
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [session 348164318] closed
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60934] closed
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60935] [c->s] OPTIONS rtsp://10.10.20.134:8554 RTSP/1.0
wyze-bridge    | CSeq: 1
wyze-bridge    | Clientchallenge: 9e26d33f2984236010ef6253fb1887f7
wyze-bridge    | Clientid: Linux_2.4_6.0.9.1235_play32_RN01_EN_586
wyze-bridge    | Companyid: KnKV4M4I/B2FjJ1TToLycw==
wyze-bridge    | Guid: 00000000-0000-0000-0000-000000000000
wyze-bridge    | Playerstarttime: [28/03/2003:22:50:23 00:00]
wyze-bridge    | Regiondata: 0
wyze-bridge    | User-Agent: RealMedia Player Version 6.0.9.1235 (linux-2.0-libc6-i386-gcc2.95)
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 D [0/0] [RTSP] [conn 10.10.20.10:60935] [s->c] RTSP/1.0 200 OK
wyze-bridge    | CSeq: 1
wyze-bridge    | Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
wyze-bridge    | Server: gortsplib
wyze-bridge    |
wyze-bridge    |
wyze-bridge    | 2021/09/08 04:58:23 I [0/0] [RTSP] [conn 10.10.20.10:60935] closed
```

**Comment by jeffreylin** [comment]
by the way - if there's a "known-good" firmware version for the Wyze Cam Pan, then let me know and I'll figure out how to flash that firmware.

**Comment by Alfiegerner** [comment]
May not be the best but I use the latest rtsp version and it works fine with this. In fact turned off the rtsp functionality as works much better with the bridge, haven't bothered looking at other firmwares.

**Comment by mrlt8** [comment]
hmm, looks like it _might_ be getting stuck on K10056SetResolvingBit which sets the resolution and bitrate.. does setting the quality to `QUALITY=SD30` help?

I believe [v4.10.6.241](https://support.wyze.com/hc/en-us/articles/360024852172-Release-Notes-Firmware) is the most recent without DTLS.

**Comment by jeffreylin** [comment]
This was fixed by flashing Wyze Cam Pan firmware version `v4.10.6.199` (the closed firmware version before `v4.10.6.241` that didn't have a release issue).

Thanks for the help everyone!
